src/ProjectNeon/Assets/Features/Effects/Events/EffectApplied.cs

```diff
@@ -0,0 +1,5 @@
+﻿public sealed class EffectApplied
```

Comment:
This needs to say which Effect was applied. Otherwise, all queued events will be dequeued, even the ones that haven't applied yet.

Reply:
@SilasReinagel Fixed, now we pass the effect.